### ISSUE TITLE
Experiment with piecemeal transition to GOA "upstreams"

### ISSUE THREAD

**Issue by kltm** [geneontology/pipeline#406]
To help with data triage, @pgaudet will experiment with changing upstreams in GOC metadata to GOA for `snapshots`

**Comment by kltm** [comment]
[Last edited 2025-07-07 -kltm]

Talking to @pgaudet, within the 172 species we'll be switching to with the GOA-produced upstream, we want to start with the set of triaged MODs, _except_:

* CGD
* GeneDB
* XenBase

Xenop is okay, if we can split it into the two species files, rather than the current grouping file.

From @pgaudet 's slides (this may have changed in the interim, the following meet the desired criteria:

This list correponds to the CURRENT CONTENT of the metadata, so it's constantly updated whenever we decide to change a file source.

- [x] dictyBase 44689
- [x] EcoCyc 83333
- [x] JaponicusDB 4897
- [x] MGI
- [x] PomBase 4896
- [x] PseudoCap 208964
- [x] RGD
- [x] SGD 559292
- [x] WB 6239
- [x] goa_chicken
- [x] goa_cow
- [x] goa_dog
- [x] goa_human
- [x] goa_pig
- [x] tair
- [x] fb
- [x] zfin


@pgaudet , I'd note that SGD and JaponicusDB also declare two species for their GAF in the metadata. That said, the there is only `JaponicusDB 4897` and `SGD 559292` in the final files. I'll proceed with just those two for now.

**Comment by kltm** [comment]
@pgaudet Starting a `snapshot` run with the updated metadata in the test PR.

**Comment by kltm** [comment]
From a discussion with @pgaudet , will add RGD as well to the testing.

**Comment by kltm** [comment]
@pgaudet As discussed: FB has been "revert" to the MOD upstream; converted MGI to GOA-sourced GAF and GPI (still want Alex to convert to our homology file as upstream.
I will wait for the current `snapshot` attempt to finish and will then run this one (likely starting Tuesday evening or Wednesday morning).

**Comment by kltm** [comment]
@pgaudet Okay, for the "current" run (not the next one as mentioned in https://github.com/geneontology/pipeline/issues/406#issuecomment-2978272646), we failed due to steep reductions in:

```
15:45:49  ERROR:sanity:Severe reduction of product for: japonicusdb
15:45:49  ERROR:sanity:Severe reduction of product for: fb
15:45:49  ERROR:sanity:Severe reduction of product for: zfin
```

I cannot push it to staging due to how early it failed. Briefly digging in:

- `japonicus`: normally covers two species (a la xenbase), so I should probably split like we did for xenbase (I believe we already discussed this). That said, there seems to have no identifiers in their GOA-sourced GAF; this is possibly a GOA issue?
- `fb`: there seems to have no identifiers in their GOA-sourced GAF; this is possibly a GOA issue? This technically doesn't matter as we're reverting them right now anyways.
- `zfin`: there seems to have no identifiers in their GOA-sourced GAF; this is possibly a GOA issue?

This is pretty much a show-stopper for these three; it does not seem (with a small random sample) to be occurring elsewhere. As Alex is still out, I am going to "revert" these three to their previous non-GOA-sourced source (adding noctua where needed) and retrying.

**Comment by pgaudet** [comment]
> japonicus: normally covers two species (a la xenbase), so I should probably split like we did for xenbase (I believe we already discussed this). That said, there seems to have no identifiers in their GOA-sourced GAF; this is possibly a GOA issue?

This should not cover 2 species. Note that there are 2 taxa - the strain (402676) - which in the last GOEx release had all annotations and the species  (4897) (associated with IBAs and with annotations provided directly by japonicusdb.

**Comment by pgaudet** [comment]
>  fb: there seems to have no identifiers in their GOA-sourced GAF; this is possibly a GOA issue? This technically doesn't matter as we're reverting them right now anyways.

Yes this is in my list of issues I will be reporting to Alex next week.

**Comment by pgaudet** [comment]
> zfin: there seems to have no identifiers in their GOA-sourced GAF; this is possibly a GOA issue?

Could be! I didn't get to that one yet. Possibly the same issues as the one causing the FB problem. Do you want to revert that source for now? If people are not consuming snapshot, maybe that's not an issue?

**Comment by kltm** [comment]
@pgaudet Updated japonicus and sgd to remove the unneeded "extra" species:
https://github.com/geneontology/go-site/commit/5291a930f9b5f733f16cb42baca1f152b59c9a2a
This should not really affect anything, except to make things a little cleaner.

**Comment by kltm** [comment]
@pgaudet a new `snapshot` is out and is up on amigo-staging.

**Comment by kltm** [comment]
From email thread with @pgaudet , revert all GPI files to original non-GOA sources.

**Comment by kltm** [comment]
@pgaudet I was able to do this for all resources, except for Xenbase, as we split the resource into two files and their provided GPI is a single file.

**Comment by kltm** [comment]
In discussion with @pgaudet and the results of https://github.com/geneontology/go-site/issues/2518, I'm striking Xenbase from the above transition list.

**Comment by pgaudet** [comment]
For chicken, just keep taxon 9031 and get rid of 
    - NCBITaxon:208524
    - NCBITaxon:208525
    - NCBITaxon:208526
    - NCBITaxon:400035

Other GOA sources can be changed to the new pipeline: human, dog, pig, bovine

**Comment by kltm** [comment]
@pgaudet Also removing from `goa_cow`:
- NCBITaxon:297284
- NCBITaxon:30523

**Comment by kltm** [comment]
Changes made and https://github.com/geneontology/pipeline/issues/406#issuecomment-2956983713 should now be up-to-date.

**Comment by kltm** [comment]
Talking to @pgaudet , "reverting" PomBase to PomBase origin.

**Comment by kltm** [comment]
@pgaudet Noting that I found the issue with the "incorrect" upstream GAFs; corrected above and rerunning `snapshot`.

**Comment by kltm** [comment]
With @pgaudet, moving tair, fb, and zfin to TODO list.

**Comment by kltm** [comment]
@pgaudet TAIR, ZFIN, and FB will be updated on the next `snapshot` out (not the one currently processing).

**Comment by kltm** [comment]
With @pgaudet pombe and japonicus can be switched--add to TODO.

**Comment by kltm** [comment]
In conversation with @pgaudet 

- [x] remove fb from noctua merge
- [x] remove zfin from noctua merge
- [x] add xenbase to noctua merge

This means that the only (non-GOEx) noctua merge is now xenbase.

**Comment by kltm** [comment]
@pgaudet This should now no longer be relevant as we fuly pivot to GOA upstream.
